src/backend/optimizer/path/indxpath.c

```diff
@@ -3529,7 +3529,7 @@ ec_member_matches_indexcol(PlannerInfo *root, RelOptInfo *rel,
 	 * generate_implied_equalities_for_column; see
 	 * match_eclass_clauses_to_index.
 	 */
-	if (index->relam == BTREE_AM_OID &&
+	if (IsIndexAccessMethod(index->relam, BTREE_AM_OID) &&
```

Comment:
This can't be simply replaced by  IsIndexAccessMethod, see the comments above.

Not everywhere could be replaced by `IsIndexAccessMethod` too,  could you check other places too?

Reply:
yes, before I made a research in all places in kernel and checked them multi times and tested them correctly.